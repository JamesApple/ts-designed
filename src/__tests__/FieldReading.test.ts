import "reflect-metadata";
import {Entity} from "..";

describe("FieldReading", function () {
  class Address extends Entity.Base {
    @Entity.Field()
    postcode: number;
  }

  class DoesNotHaveAllMethod {
    fields(): any {
      return {};
    }

    name: string;
  }

  class Person extends Entity.Base {
    @Entity.Field()
    name?: string;

    @Entity.Field({entity: Address})
    address: Address;

    @Entity.Field()
    doesNotHaveAll?: DoesNotHaveAllMethod;
  }

  it("returns all fields for the class", async function () {
    const fields = Person.fields().all();
    expect(fields[1].entityConstructor).toEqual(Address);
    expect(fields).toMatchSnapshot()
  });

  it("returns all set fields for an entity", async function () {
    const person = Person.create({
      name: "Bob",
      address: Address.create({postcode: 1})
    });
    expect(person.fields().onlySet()).toMatchSnapshot();
  });

  it("returns all unset fields for an entity", async function () {
    const person = Person.create({
      name: "Bob",
      address: Address.create({postcode: 1})
    });
    expect(person.fields().onlyUnset()).toMatchSnapshot();
    delete person.name;
    expect(person.fields().onlyUnset()).toMatchSnapshot();
  });
});
