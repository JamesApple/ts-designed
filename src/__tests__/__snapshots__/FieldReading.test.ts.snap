// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldReading returns all fields for the class 1`] = `
Array [
  Object {
    "entityConstructor": [Function],
    "fieldArrayLike": false,
    "name": "name",
    "nullable": false,
    "subFields": undefined,
  },
  Object {
    "entityConstructor": [Function],
    "fieldArrayLike": false,
    "name": "address",
    "nullable": false,
    "subFields": Array [
      Object {
        "entityConstructor": [Function],
        "fieldArrayLike": false,
        "name": "postcode",
        "nullable": false,
        "subFields": undefined,
      },
    ],
  },
  Object {
    "entityConstructor": [Function],
    "fieldArrayLike": false,
    "name": "doesNotHaveAll",
    "nullable": false,
    "subFields": undefined,
  },
]
`;

exports[`FieldReading returns all set fields for an entity 1`] = `
Array [
  Object {
    "entityConstructor": [Function],
    "fieldArrayLike": false,
    "name": "name",
    "nullable": false,
    "subFields": undefined,
  },
  Object {
    "entityConstructor": [Function],
    "fieldArrayLike": false,
    "name": "address",
    "nullable": false,
    "subFields": Array [
      Object {
        "entityConstructor": [Function],
        "fieldArrayLike": false,
        "name": "postcode",
        "nullable": false,
        "subFields": undefined,
      },
    ],
  },
]
`;

exports[`FieldReading returns all unset fields for an entity 1`] = `
Array [
  Object {
    "entityConstructor": [Function],
    "fieldArrayLike": false,
    "name": "doesNotHaveAll",
    "nullable": false,
    "subFields": undefined,
  },
]
`;

exports[`FieldReading returns all unset fields for an entity 2`] = `
Array [
  Object {
    "entityConstructor": [Function],
    "fieldArrayLike": false,
    "name": "name",
    "nullable": false,
    "subFields": undefined,
  },
  Object {
    "entityConstructor": [Function],
    "fieldArrayLike": false,
    "name": "doesNotHaveAll",
    "nullable": false,
    "subFields": undefined,
  },
]
`;
